### DISCUSSION TITLE
Write values using optolink-splitter

### DISCUSSION THREAD

**Discussion by m84v** [philippoo66/optolink-splitter#66] [Q&A]
Hi folks,
I managed setting up a Raspberry Pi with optolink-splitter to read some values via MQTT with my iobroker instance. Not too much for now, but as a first POC it is fine. Now I am interested in also setting some DPs from remote, e.g. party mode or domestic hot water temperature setpoint. I need to use vs1test branch since my boiler is only "talking" KW, but anyway. The documentation doesn't bring me any further here, since I don't know how/where to call these writing commands.
Are they passed to some of the Python scripts in the repo, and if yes how?
I would appreciate any assistance here.
Thanks in advance!

**Comment by philippoo66** [comment]
hi @m84v !

as far as I remember that shoud work the same as with vs2 - see the Wiki, [first page](https://github.com/philippoo66/optolink-splitter/wiki/010-Command-Syntax#write-requests-value), examples [here](https://github.com/philippoo66/optolink-splitter/wiki/010-Command-Syntax#examples-1)

simply publish the command as string/raw to the topic you defined as `mqtt_listen` in settings_ini ('Vitodens/cmnd' by default in the vs1 branch)

in ioBroker
receive the topic once and then click it and set/send the command
or do `setState("<topic>", "<command>");` in a js 

or use the publish function of MQTT Explorer 
or do `mosquitto_pub -h <broker_address> -p <port> -t <topic> -m "<command>"` at the command line 
or `paho_mqtt_client.publish(<topic>, <command>)` by your own py script 
or ... (sorry, with Home Assistant I'm not familiar but let me know if you need this) 

keep in mind to use the 'raw' value, so e.g. 234 for a temperature of 23.4°C scaled 0.1

**Reply by m84v** [reply]
Hi @philippoo66 
(late) thanks for reply! Just found some time again to investigate on that topic here. I am running iobroker. Procedure sounds pretty straight forward, but one thing I don't understand: What do I do if the object I want to write contains a number? Putting a string does not work since this data type is not accepted, and setting the number directly takes no effect: the original value is just re-read after the next cycle. So nothing happens. Where is the missing piece?

**Reply by FransOv** [reply]
the "w" as first character of the command instead of the "r"?

**Reply by philippoo66** [reply]
@m84v  if I'm getting you right 'your problem' is that you can not simply 'double-click' the value and do a _write_  by Home Assistant (sending the value on the polled topic). 

you need to send (publish) a command "`write;<address>;<length>;<value>`" on the topic `Vito/cmnd` (in case of `mqtt_listen = "Vito/cmnd"` else on whatever you defined there in settings_ini)

**Reply by philippoo66** [reply]
@FransOv is there a way to 'implement' such command for a dedicated topic with Home Assistant? With ioBroker I think you can select "use Converter when writing" and do some kind of formula like 
`sendTo(topic: "Vito/cmnd", message: "write;0x1234;1;{val}")`

**Reply by FransOv** [reply]
I defined an input text field in Home Assistant with a command topic and a state topic of vito/cmnd and a sensor with a state topic of vito/resp. That way I can enter a command in the input text field wich will send the command to the vito/cmnd topic on change. It will alse show the commands that are send to this topic by automations.
It looks like this:
![Screenshot_20250204_064757_Home Assistant](https://github.com/user-attachments/assets/a2d26291-829d-472b-9d48-e69101ec0fd2)

The definition in my json file to be processed by the create_entities script is:
```
{"domain":"sensor","name":"Resp","icon":"mdi:responsive"},
{"domain":"text","name":"Cmnd","icon":"mdi:gesture-tap-button","command_topic":"cmnd"},
```

**Comment by FransOv** [comment]
With Homeassistant you can use the publish and listen functions on the MQTT integration configure page.
